airflow/providers/amazon/aws/executors/ecs/utils.py

```diff
@@ -264,3 +264,16 @@ def camelize_dict_keys(nested_dict) -> dict:
         else:
             result[new_key] = nested_dict[key]
     return result
+
+
+def _deep_update(dest_dict: dict, source_dict: dict) -> dict:
+    """Deep updates dest_dict with the values from source_dict."""
+    for key, value in source_dict.items():
+        if key in dest_dict and isinstance(dest_dict[key], dict) and isinstance(value, dict):
+            # Recursively update nested dictionaries
+            _deep_update(dest_dict[key], value)
+        else:
+            # Update or add key-value pairs
+            dest_dict[key] = value
+
+    return dest_dict
```

Comment:
Is it do the same things that `airflow.utils.helpers.merge_dicts` does? 

https://github.com/apache/airflow/blob/e9ba37bb58da0e3d6739ec063f7160f50487d3b8/airflow/utils/helpers.py#L202-L214

Reply:
Ooo, looks like it! I'll update my code to use this and run the tests and ensure everything still passes.